quiche/src/recovery/mod.rs

```diff
@@ -400,6 +400,22 @@ impl Recovery {
         trace!("{} {:?}", trace_id, self);
     }
 
+    // Limit data buffered in host send buffer to avoid bufferbloat
+    // at the sender.
+    pub fn get_host_buffered(&mut self, now: Instant) -> usize {
+        let mut buffered: usize = 0;
+
+        for epoch in 0..packet::Epoch::count() {
+            for pkt in self.sent[epoch].iter() {
```

Comment:
This needs to exclude packets that have already been acked or lost as they are not immediately removed from the list, as well as include only packets that are actually "in-flight".

I think the inner loop can be replaced with something like:
```rust
            buffered += sent
                // Skip packets that have already been sent, acked or lost, or that are not in-flight.
                .filter(|p| p.in_flight && p.time_acked.is_none() && p.time_lost.is_none() && p.time_sent > now)
                .map(|x| x + p.size);
```
(not entirely sure it actually compiles though)